src/Apple/Provider.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+namespace SocialiteProviders\Apple;
+
+use Firebase\JWT\JWK;
+use Illuminate\Http\Response;
+use Illuminate\Support\Arr;
+use Laravel\Socialite\Two\InvalidStateException;
+use Laravel\Socialite\Two\ProviderInterface;
+use Lcobucci\JWT\Parser;
+use Lcobucci\JWT\Signer\Rsa\Sha256;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider implements ProviderInterface
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    const IDENTIFIER = 'APPLE';
+
+    private const URL = 'https://appleid.apple.com';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = [
+        'name',
+        'email',
+    ];
+
+    /**
+     * The separating character for the requested scopes.
+     *
+     * @var string
+     */
+    protected $scopeSeparator = ' ';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase(self::URL.'/auth/authorize', $state);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return self::URL.'/auth/token';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getCodeFields($state = null)
+    {
+        $fields = [
+            'client_id'     => $this->clientId,
+            'redirect_uri'  => $this->redirectUrl,
+            'scope'         => $this->formatScopes($this->getScopes(), $this->scopeSeparator),
+            'response_type' => 'code',
+            'response_mode' => 'form_post',
+        ];
+
+        if ($this->usesState()) {
+            $fields['state'] = $state;
+            $fields['nonce'] = time().'-'.$state;
```

Comment:
If this needs to be a unique string you could rely on `Illuminate\Support\Str::uuid()` it is a dependency of Socialite anyway.

Reply:
Thank You @jmarcher. We replaced time with in-built `Illuminate\Support\Str::uuid()`.